lib/routes/account.js

```diff
@@ -581,41 +578,54 @@ module.exports = function (
         }
 
         function checkEmailAndPassword () {
-          // Session token verification is only enabled for certain users during phased rollout.
-          //
-          // If the user went through the sigin-unblock flow, they have already verified their email.
-          // No need to also require confirmation afterwards.
-          //
-          // Even when it is enabled, we only do the email challenge if:
-          //  * the request wants keys, since unverified sessions are fine to use for e.g. oauth login.
-          //  * the email is verified, since content-server triggers a resend of the verification
-          //    email on unverified accounts, which doubles as sign-in confirmation.
-          //  * the login is flagged that it can be bypassed
-
-          // Check to see if this login can bypass sign-in confirmation. Current scenarios include
-          //  * User has already logged in from this ip address and verified the sign-in
-          let bypassSiginConfirmation = features.canBypassSiginConfirmation(emailRecord.email, securityEventVerified, securityEventRecency)
-          if (bypassSiginConfirmation) {
-            log.info({
-              op: 'Account.ipprofiling.seenAddress',
-              uid: emailRecord.uid.toString('hex')
-            })
```

Comment:
I unwound the logic here into a couple of independent tests.  It's more verbose but IMHO easier to follow.  The diff has made a mess of it though, you might need to just view the new version of the file to see what it looks like.